lua/impatient.lua

```diff
@@ -223,17 +225,29 @@ local function load_from_cache(name)
 end
 
 function M.save_cache()
-  if M.dirty then
-    log('Updating cache file: %s', M.path)
-    local f = io.open(M.path, 'w+b')
-    f:write(mpack.pack(M.cache))
-    f:flush()
+  if not M.dirty then return end
+
+  if _G.__luacache_use_sqlite then
+    log('Updating cache db')
+    cache:save(mpack.pack(M.cache))
     M.dirty = false
+    return
   end
+
```

Comment:
Can we refactor this, so the default cache uses the same API as the sqlite cache? Though keep the default cache in this module for now, so it doesn't need to be `require`'d

Something like:

```lua
local cache = {
  init = function(data) ...end,
  save = function() ... end,
  clear = function() ... end,
}
```